packages/amazonq/test/e2e_new/amazonq/framework/generalHelper.ts

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { By, WebviewView, WebElement } from 'vscode-extension-tester'
+import { until } from 'selenium-webdriver'
+
+/* Waits for an element (or multiple elements) to appear based on the parameters
+
+Logic:
+The function utilizes the Selenium wait driver. We can call that driver from our
+WebviewView but it can also be called on parts of the VSCode Editor that are not
+part of the WebviewView.
+
+(TO-DO: create a more general function that can be called on any part of the VSCode
+Editor. Will do when a use case appears for it.*/
+
+export async function waitForElement(
+    webview: WebviewView,
+    locator: By,
+    multiple: true,
+    timeout?: number
+): Promise<WebElement[]>
+export async function waitForElement(
+    webview: WebviewView,
+    locator: By,
+    multiple?: false,
+    timeout?: number
+): Promise<WebElement>
+export async function waitForElement(
+    webview: WebviewView,
+    locator: By,
+    multiple = false,
+    timeout = 15000
+): Promise<WebElement | WebElement[]> {
+    const driver = webview.getDriver()
+    await driver.wait(until.elementsLocated(locator), timeout)
+    return multiple ? await webview.findWebElements(locator) : await webview.findWebElement(locator)
+}
+
+/* General function for finding WebElement by their text content
```

Comment:
worth noting that it only returns the first, and ignores the rest (based on my understanding of the implementation).